src/Base/Imp/Desktop/EmbeddedResourcesDllHandler.cs

```diff
@@ -33,7 +33,8 @@ public static void LoadEmbeddedDll(string dllName, string resourceName)
                 {
                     // The temporary folder holds one or more of the temporary DLLs
                     // It is made "unique" to avoid different versions of the DLL or architectures.
-                    var tempFolder = String.Format("{0}.{1}.{2}", an.Name, an.ProcessorArchitecture, an.Version);
+                    var pa = RuntimeInformation.ProcessArchitecture == Architecture.X64 ? "x64" : "x32";
+                    var tempFolder = string.Format("{0}.{1}.{2}", an.Name, pa, an.Version);
```

Comment:
`RuntimeInformation.ProcessArchitecture.ToString()` ([Enum.ToString](https://learn.microsoft.com/en-us/dotnet/api/system.enum.tostring?view=net-7.0)) should work here? This would make L36 obsolete.